Firestore/Source/API/FIRDocumentSnapshot+Internal.h

```diff
@@ -17,23 +17,33 @@
 #import "FIRDocumentSnapshot.h"
 
 #include "Firestore/core/src/firebase/firestore/api/document_snapshot.h"
+#include "Firestore/core/src/firebase/firestore/api/snapshot_metadata.h"
 #include "Firestore/core/src/firebase/firestore/model/document_key.h"
 
 @class FIRFirestore;
 @class FSTDocument;
 
+using firebase::firestore::api::DocumentSnapshot;
```

Comment:
Note that putting these kinds of `using` statements in the Objective-C header is controversial, but I'm putting it out there because it's getting tedious to repeat these hugely long names.

It's particularly bad for templates, where e.g. `StatusOrCallback<DocumentSnapshot>` has to be spelled `firebase::firestore::util::StatusOrCallback<firebase::firestore::api::DocumentSnapshot>`.

I propose we allow this for non-public headers only. Public headers have to be Objective-C so these C++ types aren't valid there anyway.

Reply:
I think this is okay. We never include those headers in our public headers, so I don't think those `using` statements should leak out? The only concern is that we sometimes have same class names that are only differentiated by the namespace. However, this should result in a compile-time error, so it doesn't seem to be a big issue.

Reply:
Agreed and we actually need to fix that differing-by-namespace issue anyway because it trips up the cocoapod build generating a warning about a duplicate file being omitted.

Reply:
FWIW, the style guide (if I twist it enough) even says this is fine. "However, local convenience aliases are fine in function definitions, private sections of classes, explicitly marked **internal namespaces**, and in .cc files:"

One could argue that these internal obj-c headers are kinda an internal namespace.